keps/sig-scheduling/4328-affinity-based-eviction/README.md

```diff
@@ -0,0 +1,371 @@
+
+- Inspect the nodeAffinity.RequiredDuringSchedulingRequiredDuringExecution configuration.
+- Check the corresponding metrics(details in SLIs part below).
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Events
+  - Event Reason: NodeAffinityEviction
+- [x] API .status
+  - Condition reason: DeletionByAffinityEvictionController 
+- [x] Other (treat as last resort)
+  - Details: Pod with `RequiredDuringSchedulingRequiredDuringExecution` must meet affinity rule during both the scheduling phase and the execution phase.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name:
+    - pod_deletions_node_affinity_total(counts the number of Pods deleted because of nodeAffinity since this contoller start)
+    - pod_deletions_node_affinity_duration_seconds(Latency, in seconds, between the time when a node's label has been changed and the pod been deleted via AffinityEvictionController)
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Two new metrics will be added.
+
+### Dependencies
+
+N/A
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+- There will be calls to PATCH pod status.
+- There will be calls to delete pod.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+It increasing the size of `NodeAffinity` a little bit.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+If `RequiredDuringSchedulingRequiredDuringExecution` field is defined, this feature will slightly increase the pod scheduling time and increase the possibility of pods being evicted.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+The feature will not work.
+
+###### What are other known failure modes?
+
+N/A
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+- 2023-12-13: Initial draft KEP
+
+## Drawbacks
+
+N/A
+
+## Alternatives
```

Comment:
also implementing it in the kubelet is an alternative. Please list it and explain why it's not a good idea.

Reply:
Added... Not sure if it's sufficient.